strawberryfields/gbs/points.py

```diff
@@ -66,3 +68,34 @@ def kernel(R: np.ndarray, sigma: float) -> np.ndarray:
         K (array): the kernel matrix
     """
     return np.exp(-(scipy.spatial.distance.cdist(R, R)) ** 2 / 2 / sigma ** 2)
+
+
+def perpp(R: np.ndarray, sigma: float, n_mean: int, ns: int) -> np.ndarray:
+    """This function generates permanental point process (PerPP) samples with thermal states.
```

Comment:
```suggestion
    """Sample subsets of points using the permanental point process.

    Points are encoded through a radial basis function kernel, provided in :func:`kernel`,
    that is a function of the distance between pairs of points. Subsets of points are sampled
    with probabilities that are proportional to the permanent of the *sub*matrix of the kernel
    selected by those points.

    This permanental point process is likely to sample points that are clustered together
    :cite:`soran2019`. It can be realised using a variant of Gaussian boson sampling with thermal
    states as input.
```
Feel free to reword!
Note that, you need to add a citation to `doc/references.bib`.